src/Lean/Environment.lean

```diff
@@ -616,10 +619,12 @@ partial def importModules (imports : List Import) (opts : Options) (trustLevel :
   try
     importingRef.set true
     let (_, s) ← importMods imports |>.run {}
-    -- (moduleNames, mods, regions)
+    let mut numConsts := 0
+    for mod in s.moduleData do
+      numConsts := numConsts + mod.constants.size
     let mut modIdx : Nat := 0
-    let mut const2ModIdx : HashMap Name ModuleIdx := {}
-    let mut constants : ConstMap := SMap.empty
+    let mut const2ModIdx : HashMap Name ModuleIdx := Std.mkHashMap (nbuckets := numConsts)
+    let mut constants : ConstMap := SMap.empty (nbuckets₁ := numConsts)
```

Comment:
@leodemoura This was a low-hanging fruit for significantly speeding up this part of importing by removing the need for resizing the hashmaps. I want to try further simple optimizations of the hashmap such as enforcing a "load factor" (usually 0.75), but not in this PR.